### PR TITLE
fix: add manage-sources-list-d CLI option (LP:2077360) ; refactor: optparse to argparse for CLI options

### PR DESCRIPTION
**Pull request by HJK-X**
Refactor optparse to argparse. Should be mostly a drop-in replacement except for positional arguments (which were previously ignored). 

https://bugs.launchpad.net/landscape-client/+bug/2077360

To create similar functionality for `--manage-sources-list-d` as described in the report, landscape-config will parse the terminal command with an optional argument for t/f values.

### CODE REVIEW

File: landscape/lib/tests/test_config.py [hunk 1]
```diff
@@ -159,6 +172,118 @@ def test_get_config_filename_precedence(self):
         self.config.config = explicit_filename
         self.assertEqual(self.config.get_config_filename(), explicit_filename)
 
+    # argparse CLI parsing
+
+    def test_parse_default_positional_argument(self):
```
Comment (srunde3) [inline landscape/lib/tests/test_config.py:177]: Could you please add docstrings to these tests?

**Review by srunde3 — COMMENTED** [review]
Overall looks good; a couple inline asks

File: landscape/client/configuration.py [hunk 2]
```diff
@@ -199,68 +200,76 @@ def make_parser(self):
...
-        parser.add_option(
+        parser.add_argument(
             "--include-manager-plugins",
             metavar="PLUGINS",
             default="",
             help="A comma-separated list of manager plugins "
             "to enable in addition to the defaults.",
         )
-        parser.add_option(
+        parser.add_argument(
```
[the remark this answers is not in the document]
Comment (HJK-X) [reply landscape/client/configuration.py:217]: Yep, so a user can do 
--manage-sources-list-d false
--manage-sources-list-d true
--manage-sources-list-d

the last one will also set it to true

File: landscape-client.conf [hunk 3]
```diff
@@ -2,7 +2,7 @@
 bus = session
 computer_title = John's PC
 account_name = onward
-registration_key =
+registration_key =
```
[the remark this answers is not in the document]
Comment (HJK-X) [reply landscape-client.conf:5]: Yeah, I think this is probably testing

**Review by srunde3 — APPROVED** [review]
LGTM :shipit:

**Pull request merged**